### PR TITLE
Http hook

### LINKED ISSUE #114: Move hookcontroller into http function trigger
I would like to move the website landing pages to static hosting and the controller into a stand alone Azure function.

See https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook

We should add some tests that the hook is doing it's job too

### PR DESCRIPTION
**Pull request by dabutvin**
fixes #114

### CODE REVIEW

File: Test/Test.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp2.1</TargetFramework>
     <IsPackable>false</IsPackable>
+    <LangVersion>7.2</LangVersion>
```
Comment (dlemstra) [inline Test/Test.csproj]: You could use `latest` here instead.

File: Test/data/hooks/installation_repositories_created.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "id": 541,
+    "account": {
+      "login": "dabut",
+      "id": 12345,
+      "node_id": "MDQ6VXNlcjQyODI3NTAw",
+      "avatar_url": "https://avatars2.githubusercontent.com/u/12345?v=4",
+      "gravatar_id": "",
+      "url": "https://api.github.com/users/dabut",
+      "html_url": "https://github.com/dabut",
+      "followers_url": "https://api.github.com/users/dabut/followers",
```
Comment (dlemstra) [inline Test/data/hooks/installation_repositories_created.json]: Should this be `dabutvin` instead of `dabut`?

File: WebHook/WebHookFunction.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    {
+                        AccountType = hook.marketplace_purchase.account.type,
+                        PlanId = hook.marketplace_purchase.plan.id,
+                        SenderId = hook.sender.id,
+                        SenderLogin = hook.sender.login,
+                    }));
+
+                    return "purchased";
+                case "cancelled":
+                    await marketplaceTable.DropRow(hook.marketplace_purchase.account.id.ToString(), hook.marketplace_purchase.account.login);
```
Comment (dlemstra) [inline WebHook/WebHookFunction.cs]: Looks a bit odd that you need the `ToString` in the delete but not in the insert? Maybe you could do that inside the `DropRow` function?

**Comment by dabutvin** [comment]
thanks for the review 😎

**Pull request merged**